From 774e9f6b8afa3a7dfef3c721ab6008823d196558 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Marie?= <semarie@users.noreply.github.com>
Date: Sun, 18 Jan 2015 13:53:37 +0100
Subject: [PATCH] libbacktrace under openbsd need filename

passing NULL to libbacktrace under openbsd results:
 - libbacktrace could not find executable to open
 - failed to read executable information

should solve 'run-pass/backtrace.rs' failure, when 'os::self_exe_name'
will be implemented.
---
 src/libstd/sys/unix/backtrace.rs | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libstd/sys/unix/backtrace.rs b/src/libstd/sys/unix/backtrace.rs
index 7164931..ae28076 100644
--- a/src/libstd/sys/unix/backtrace.rs
+++ b/src/libstd/sys/unix/backtrace.rs
@@ -318,7 +318,8 @@ fn print(w: &mut Writer, idx: int, addr: *mut libc::c_void) -> IoResult<()> {
         static mut LAST_FILENAME: [libc::c_char; 256] = [0; 256];
         if !STATE.is_null() { return STATE }
         let selfname = if cfg!(target_os = "freebsd") ||
-                          cfg!(target_os = "dragonfly") {
+                          cfg!(target_os = "dragonfly") ||
+                          cfg!(target_os = "openbsd") {
             os::self_exe_name()
         } else {
             None
-- 
2.2.2

